x/gamm/pool-models/balancer/pool_suite_test.go

```diff
@@ -823,25 +953,23 @@ func (suite *KeeperTestSuite) TestCalcJoinPoolShares() {
 
 			// system under test
 			sut := func() {
-				shares, liquidity, err := pool.CalcJoinPoolShares(suite.Ctx, tc.tokensIn, tc.swapFee)
+				preJoinAssets := pool.GetTotalPoolLiquidity(suite.Ctx)
+				shares, err := pool.JoinPoolNoSwap(suite.Ctx, tc.tokensIn, tc.swapFee)
```

Comment:
q: if the caller provides uneven amounts to `JoinPoolNoSwap` such as `25000` and `50000`, how can the caller tell how many tokens were joined and how many are remaining? We might want to consider having an extra return value to represent either the "tokens joined" or "remaining tokens".

If you agree, let's make an issue for this please

Reply:
Yeah good point – I don't think this is too critical of a change given how `JoinPoolNoSwap` _does_ return tokens joined at the keeper level, but I think it currently does so by taking a bit of a shortcut so I think this is worth adding for code hygiene